### DISCUSSION TITLE
Can't get RNG to work on a ST Nucleo board with support for entropy driver

### DISCUSSION THREAD

**Discussion by EsperLiu** [zephyrproject-rtos/zephyr#55915] [Q&A] [answered]
I'm on the latest versions of everything and let's say I want to generate a random number with a [ST Nucleo L552ZE Q](https://docs.zephyrproject.org/latest/boards/arm/nucleo_l552ze_q/doc/nucleol552ze_q.html#st-nucleo-l552ze-q), which I see has an entropy driver supported. 

Let's say I have the following code and use `CONFIG_TEST_RANDOM_GENERATOR=y` in my `prj.conf`: 

```
#include <zephyr/kernel.h>
#include <zephyr/random/rand32.h>

void main(void)
{
	int my_number = sys_rand32_get();
	printk("My random number is: %i \n", my_number);
}
```

This program complies, but I always get the same number printed out. 

If I replace the test random number generator with anything else, i.e. use `CONFIG_XOSHIRO_RANDOM_GENERATOR=y`, I get 

```
home/username/zephyrproject/build/zephyr/include/generated/syscalls/rand32.h:32: undefined reference to `z_impl_sys_rand32_get'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/username/zephyrproject/build
```
I must have missed something here because the documentation explicitly says the RNG feature on my ST board is supported. How should I use Zephyr's RNG?

Thanks for any help in advance.

**Comment by yashi** [comment, marked as answer]
`CONFIG_TEST_RANDOM_GENERATOR` is not a real one.  See https://docs.zephyrproject.org/latest/services/crypto/random/index.html#kconfig-options

To use a RNG on a Nucleo, you can do some like

```
west build -b nucleo_f429zi zephyr/samples/hello_world -- -DCONFIG_ENTROPY_GENERATOR=y
```

And you get

```
*** Booting Zephyr OS build zephyr-v3.3.0-1222-gc49a6a0b0cd9 ***                  
Hello World! nucleo_f429zi                                                        
*** Booting Zephyr OS build zephyr-v3.3.0-1222-gc49a6a0b0cd9 ***                  
My random number is: 1715436118                                                   
*** Booting Zephyr OS build zephyr-v3.3.0-1222-gc49a6a0b0cd9 ***
My random number is: -2078358921 
*** Booting Zephyr OS build zephyr-v3.3.0-1222-gc49a6a0b0cd9 ***
My random number is: -1229691571 
```
